### DISCUSSION TITLE
Questions about modifying highlight behaviour

### DISCUSSION THREAD

**Discussion by ambadyanands** [ggandor/lightspeed.nvim#163] [Q&A]
As shown in [this image](https://raw.githubusercontent.com/ggandor/lightspeed.nvim/media/quick_example_2.png) in the README, there are three types of matches shown when we type `s{char}`:
1. Directly reachable match (that needs needs just one more character press).
2. Normal label (that needs needs two more character press).
3. Shortcut label (that needs one OR two more character press).

The 1st one highlights the already typed character and the next one to be typed. May I please know the design choice behind highlighting the already typed character? Would it be possible to disable highlighting the already typed character? (i.e. `m` in the screenshot mentioned).

In the 2nd case, the normal labelled character cannot be typed directly without figuring out the character just before the labelled one which isn't highlighted. IMO it would be more easier if both characters were highlighted, would it be possible to do the same? 

These questions won't matter much if we type two characters after `s`, which is the usual case I believe. And maybe it's just a matter of not practicing it enough. But still, I believe it'd be better to alter the highlighting behaviour as mentioned above to reduce the cognitive load in case of `s{char}`. The idea is to always highlight [only] the characters that need to be typed so that we don't have to think much about the kind of match and the meaning of highlight. Instead, if it's highlighted just type and jump.

Thanks and lots of love for this amazing plugin.

**Comment by ggandor** [comment]
> The idea is to always highlight [only] the characters that need to be typed so that we don't have to think much about the kind of match and the meaning of highlight. Instead, if it's highlighted just type and jump

No, that's not exactly the aim - in that case you would always have to be alert, and _react_ to some change, after _each_ keypress. We want the exact opposite. I just copied this comment directly from the `lightspeed.fnl` file, maybe this will help you understand my thought process and the desing approach:

; The first principle of Lightspeed's interface is that after a beacon appears
; on screen, it should not change in appearance unless strictly necessary. That
; is because the brain has to be given as much time as possible to familiarize
; itself with the beacon, and any sudden change would only be confusing and
; ultimately counterproductive.
; Therefore, labels will not change into "shortcuts" after the second input,
; even though, in a way, it would be logical. If a label was part of an
; overlapped beacon, it will not get transformed back into a regular label when
; the first column gets uncovered. For the same reason, the masked characters
; will not disappear before the labels in the second round. The look of a beacon
; only changes with group switching, when its active/passive or
; primary/secondary state changes.

> In the 2nd case, the normal labelled character cannot be typed directly without figuring out the character just before the labelled one which isn't highlighted. IMO it would be more easier if both characters were highlighted, would it be possible to do the same?

The idea is that you probably have already loaded those two pattern characters (in the match) into your brain, when you chose the target and pressed `s` - highlighting them is redundant. If we skip that, and only highlight the directly reachable (unlabeled) ones, we can distinguish between type 1 and type 2/3 matches easily. Since the _main_ question after the first input is, "do I need to a enter label or not?"

Feel free to discuss this further, I'm always interested in user's experiences and opinions, sometimes the feedback is really valuable.

> Thanks and lots of love for this amazing plugin.

<3
